tedana/model/fit.py

```diff
@@ -496,51 +485,64 @@ def gscontrol_raw(catd, optcom, n_echos, ref_img, dtrank=4):
     return dm_catd, dm_optcom
 
 
-def spatclust(data, mask, csize, thr, ref_img, infile=None, dindex=0,
-              tindex=0):
+def spatclust(img, min_cluster_size, threshold=None, index=None, mask=None):
     """
-    Thresholds and spatially clusters `data`
+    Spatially clusters `img`
 
     Parameters
     ----------
-    data : (S x T) array-like
-        Input data array
-    mask : (S,) array-like
-        Boolean mask array
-    csize : int
-        Size of cluster (in voxels) to retain
-    thr : float
-        Value to threshold image at before clustering
-    ref_img : str or img_like
-        Reference image to dictate how outputs are saved to disk
-    infile : str, optional
-        Path to file that should be used for clustering instead of `data`.
-        Default: None
-    dindex : int, optional
-        Index of data (2nd dimension) to use for clustering. Default: 0
-    tindex : int, optional
-        Index of data (2nd dimension) to use for thresholding. Default: 0
+    img : str or img_like
+        Image file or object to be clustered
+    min_cluster_size : int
+        Minimum cluster size (in voxels)
+    threshold : float, optional
+        Whether to threshold `img` before clustering
+    index : array_like, optional
+        Whether to extract volumes from `img` for clustering
+    mask : (S,) array_like, optional
+        Boolean array for masking resultant data array
 
     Returns
     -------
-    clustered : (S x T) np.ndarray
-        Boolean array indicated data samples to be retained after clustering
+    clustered : np.ndarray
+        Boolean array of clustered (and thresholded) `img` data
     """
 
-    if infile is None:
-        data = data.copy()
-        data[data < thr] = 0
-        infile = utils.filewrite(utils.unmask(data, mask), '__clin.nii.gz', ref_img, gzip=True)
-
-    addopts = ''
-    if data is not None and data.squeeze().ndim > 1 and dindex + tindex == 0:
-        addopts = '-doall'
-    else:
-        addopts = '-1dindex {0} -1tindex {1}'.format(str(dindex), str(tindex))
-
-    cmd_str = '3dmerge -overwrite {0} -dxyz=1 -1clust 1 {1:d} ' \
-              '-1thresh {2:.02f} -prefix __clout.nii.gz {3}'
-    os.system(cmd_str.format(addopts, int(csize), float(thr), infile))
+    img = niimg.copy_img(check_niimg(img, atleast_4d=True))
+
+    # make sure voxel sizes are 1mm^3
```

Comment:
What happens if this is not true ?

Reply:
It's not so much "make sure" as it is "temporarily set voxel sizes to 1mm^3". That way the `min_cluster_size` argument represents the number of voxels we want in a cluster rather than the actual size of the cluster in mm^3. This was originally accomplished using `3dmerge -dxyz=1`; `nilearn.regions.connected_regions` does't have a comparable parameter.

We could instead do some calculation of the actual voxel volume, but I wasn't sure how rounding errors would be treated. Moreover, temporarily resetting the affine should be okay because it (the affine) is discarded when the clustered data is loaded back in to memory: 
```
clustered = utils.load_image(niimg.concat_imgs(clout).get_data()) != 0
```

Reply:
Ah, got it. Sorry, I misinterpreted the code comment ! I think that method totally makes sense. Would it be possible to update the comment ?

Reply:
Definitely! Apologies for being so minimalist with them. I'll try to be a bit more descriptive in the future :smile: